### PR TITLE
reorganize workspace

### PR DESCRIPTION
**Pull request by minghuaw**
Reorganizing the workspace for a cleaner structure. The future `mlx-nn` crate and the ergonomics-focused crate would likely depend on the `mlx-rs` crate itself, having a "flat" structure would probably help organizing the dependencies better

### CODE REVIEW

**Comment by minghuaw** [comment]
We probably need to change the GH action for building the docs too

File: mlx-rs/Cargo.toml
Comment (dcvz) [inline mlx-rs/Cargo.toml:8]: not a big deal but these can be grabbed from workspace Cargo.toml

**Review by dcvz — APPROVED** [review]

File: mlx-rs/Cargo.toml
Comment (minghuaw) [reply mlx-rs/Cargo.toml:8]: Didn't see this before I hit the squash and merge button lol. We could do this later then.

**Pull request merged**
